File: frontend/src/lib/samba/playautoAlias.ts
export function normalizePlayautoAliasCode(value: string | null | undefined): string {
  const trimmed = String(value || '').trim()
  if (!trimmed) return ''
  const upper = trimmed.toUpperCase()
  if (/^\d+\.0+$/.test(upper)) return upper.replace(/\.0+$/, '')
  return upper
}

export function parsePlayautoAliasEntry(value: string | null | undefined): { code: string; alias: string } {
  const raw = String(value || '').trim()
  if (!raw) return { code: '', alias: '' }
  const match = raw.match(/^(.*?)\s*[-\u2010-\u2015\u2212]\s*(.+)$/)
  if (!match) return { code: normalizePlayautoAliasCode(raw), alias: '' }
  return {
    code: normalizePlayautoAliasCode(match[1]),
    alias: String(match[2] || '').trim(),
  }
}

export function formatPlayautoAliasEntry(code: string | null | undefined, alias: string | null | undefined): string {
  const trimmedCode = String(code || '').trim()
  const trimmedAlias = String(alias || '').trim()
  if (!trimmedCode) return trimmedAlias
  if (!trimmedAlias) return trimmedCode
  return `${trimmedCode}-${trimmedAlias}`
}

File: backend/backend/domain/samba/order/playauto_alias.py
import re
from typing import Any


def normalize_playauto_alias_code(value: Any) -> str:
    raw = str(value or "").strip()
    if not raw:
        return ""
    upper = raw.upper()
    if re.fullmatch(r"\d+\.0+", upper):
        return re.sub(r"\.0+$", "", upper)
    return upper


def parse_playauto_alias_entry(value: Any) -> tuple[str, str]:
    raw = str(value or "").strip()
    if not raw:
        return "", ""
    match = re.match(r"^(.*?)\s*[-\u2010-\u2015\u2212]\s*(.+)$", raw)
    if not match:
        return normalize_playauto_alias_code(raw), ""
    code = normalize_playauto_alias_code(match.group(1))
    nick = str(match.group(2) or "").strip()
    return code, nick
